app/controllers/api_controller/parser.rb

```diff
@@ -41,7 +40,7 @@ def validate_optional_collection_classes
           raise BadRequestError, "Management of #{cname} is unsupported for the Provider class"
         end
       end
-      @collection_klasses[:providers] = "Provider"
+      @collection_klasses[:providers] = Provider
```

Comment:
why this change ?   the access method collection_class constantizes these so this would fail.